### PR TITLE
Spring practice

### PR DESCRIPTION
**Pull request by sjlarson92**
created dog API

### CODE REVIEW

File: java/practice/apiPractice/src/main/java/com/springDemo/apiPractice/ApiPracticeApplication.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.springDemo.apiPractice;
+
+import com.springDemo.apiPractice.controller.DogController;
```
Comment (lgc13) [inline java/practice/apiPractice/src/main/java/com/springDemo/apiPractice/ApiPracticeApplication.java]: unused import could be removed, right?

File: java/practice/apiPractice/src/main/java/com/springDemo/apiPractice/controller/DogController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        Dog defaultDog = dogService.getDefaultDog();
+        Dog dog1 = dogService.createNewDog(firstName, lastName, age, goodDog, breed);
+        return dogService.compareDogs(defaultDog, dog1);
+    }
+
+    @RequestMapping("/images")
+    private DogImageResponse getDogImages(){
+        final String uri = "https://dog.ceo/api/breeds/image/random";
+        RestTemplate restTemplate = new RestTemplate();
+        DogImageResponse result = restTemplate.getForObject(uri, DogImageResponse.class);
```
Comment (lgc13) [inline java/practice/apiPractice/src/main/java/com/springDemo/apiPractice/controller/DogController.java]: you could return this result directly, right?

File: java/practice/apiPractice/src/main/java/model/Dog.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package com.springDemo.apiPractice.model;
+
+public class Dog {
+    public enum Breed {
```
Comment (lgc13) [inline java/practice/apiPractice/src/main/java/model/Dog.java]: can this be private?

File: java/practice/apiPractice/src/main/java/model/DogCompareResponse.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package com.springDemo.apiPractice.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DogCompareResponse {
+    String oldestDogName;
```
Comment (lgc13) [inline java/practice/apiPractice/src/main/java/model/DogCompareResponse.java]: cam both of these fields be set to private?

File: java/practice/apiPractice/src/main/java/com/springDemo/apiPractice/ApiPracticeApplication.java (hunk 1 above)
Comment (sjlarson92) [reply java/practice/apiPractice/src/main/java/com/springDemo/apiPractice/ApiPracticeApplication.java]: removed

File: java/practice/apiPractice/src/main/java/com/springDemo/apiPractice/controller/DogController.java (hunk 2 above)
Comment (sjlarson92) [reply java/practice/apiPractice/src/main/java/com/springDemo/apiPractice/controller/DogController.java]: I want to keep it separate so I can use it as reference for later

File: java/practice/apiPractice/src/main/java/model/Dog.java (hunk 3 above)
Comment (sjlarson92) [reply java/practice/apiPractice/src/main/java/model/Dog.java]: yes and it still works

File: java/practice/apiPractice/src/main/java/model/DogCompareResponse.java (hunk 4 above)
Comment (sjlarson92) [reply java/practice/apiPractice/src/main/java/model/DogCompareResponse.java]: yes and done

**Pull request merged**